api/v1alpha1/edgeconfig_types.go

```diff
@@ -0,0 +1,202 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// EdgeConfigSpec defines the desired state of EdgeConfig
+type EdgeConfigSpec struct {
+	// Definition of the custom playbook workload to execute
+	EdgePlaybook *EdgePlaybookSpec `json:"edgePlaybook,omitempty"`
+
+	// The deviceID list on which the playbook should be executed.
+	// Necessary to execute the playbook on a devices that don't belong to any group.
+	//+optional
+	DeviceIDs []string `json:"deviceIDs,omitempty"`
+}
+
+// EdgeConfigStatus defines the observed state of EdgeConfig
+type EdgeConfigStatus struct {
+	EdgePlaybookStatus *EdgePlaybookStatus `json:"edgePlaybookStatus,omitempty"`
+}
+
+// EdgePlaybookSpec defines the desired state of EdgePlaybook
+type EdgePlaybookSpec struct {
+	// The ansible playbook command to execute
+	AnsiblePlaybookCmd *AnsiblePlaybookCmd `json:"ansiblePlaybookCmd,omitempty"`
+}
+
+// EdgePlaybookStatus defines the observed state of EdgePlaybook
+type EdgePlaybookStatus struct {
+	Condition                 []EdgePlaybookCondition `json:"condition,omitempty"`
+	LastSeenTime              metav1.Time             `json:"lastSeenTime,omitempty"`
+	LastSyncedResourceVersion string                  `json:"lastSyncedResourceVersion,omitempty"`
+}
+
+type AnsiblePlaybookCmd struct {
+	// username who execute the playbook
+	User string `json:"user,omitempty"`
+	// the ansible's playbooks list with execution priority
+	// +kubebuilder:validation:MinProperties=1
+	PlaybooksPriorityMap map[string]Playbook `json:"playbooksPriorityMap,omitempty"`
```

Comment:
@masayag regarding your previuous comment
> if we expect a list of playbooks, is there any importance for the order of execution or any dependency? (e.g. if 1/2 fails, should the second one be executed?)

I create a map in which the key is the priority of execution.
I'd like to use an integer as key but:
```
> make manifests
/home/gloria/repos/flotta-operator/api/v1alpha1/edgeconfig_types.go:60:27: map keys must be strings, not int
```

IS there a better way to handle order of execution?

Reply:
I think that if u need an order then u should use a slice.

Reply:
@ydayagi yes it's a possibility! my concern is that, in that case, there is no "explicit" execution priority. 
Can be this a problem? WDYT?